plugin/config_test.go

```diff
@@ -238,6 +239,206 @@ pipeline:
 	}
 }
 
+type PluginOutputIDTestCase struct {
+	Name          string
+	PluginConfig  pipeline.Config
+	ExpectedOpIDs map[string][]string
+}
+
+func TestPluginOutputIDs(t *testing.T) {
```

Comment:
Really nice tests. This is a powerful little harness, and I think we should expand on it even more because it can really lock in the functionality.

Can you think of any other test cases?

Reply:
One that I'd like to see is having `my_plugin` feed into another instance of `my_plugin`.

For now, this will require specifying an `id` for each instance of the plugin. Once autogenerated IDs are implemented, we can validate that these IDs can be removed.

```yaml
pipeline: 
  - type: noop 
  - type: my_plugin 
  - type: my_plugin
    id: my_plugin1 # TODO remove when autogen IDs implemented 

# my_plugin 
pipeline: 
  - type: noop 
  - type: noop 
    id: noop1 # TODO remove when autogen IDs implemented 
    output: {{ .output }}
```

Reply:
Another would be something like `operator -> plugin -> operator -> plugin`

Reply:
would a plugin within another plugin be applicable

Reply:
Also, I added that one and another one.

Reply:
> would a plugin within another plugin be applicable

Yes, good point. Let's have some tests

Reply:
Added the additional test.